nix_update/version/github.py

```diff
@@ -22,31 +33,59 @@ def version_from_entry(entry: Element) -> Version:
     return Version(unquote(url.path.split("/")[-1]))
 
 
+def _dorequest(url: ParseResult, feed_url: str) -> str:
+    request = urllib.request.Request(feed_url)
+
+    try:
+        netrccreds = netrc.netrc().authenticators(url.netloc)
+        if netrccreds is not None:
+            info("using netrc file")
+            encoded = f"{netrccreds[0]}:{netrccreds[2]}".encode()
+            encodedcreds = base64.b64encode(encoded).decode()
+            request.add_header("Authorization", f"Basic {encodedcreds}")
+    except FileNotFoundError:
+        pass
+
+    return urllib.request.urlopen(request).read()
+
+
 def fetch_github_versions(url: ParseResult) -> list[Version]:
-    if url.netloc != "github.com":
+    # sourcehut and github share the same /archive/xxx.tar.gz path structure
+    if url.netloc == "git.sr.ht":
```

Comment:
I find it a bit odd that we are mixing sourcehut and github here. Why is this the case?

Reply:
Ok. Sourcehut is also self-hostable. So I assume we might have this issue later again. Could we filter on their usernames? `some.host/~name`. This might be more robust.